addons/website/static/src/js/content/snippets.animation.js

```diff
@@ -653,15 +701,34 @@ registry.mediaVideo = publicWidget.Widget.extend({
         // TODO: this code should be refactored to make more sense and be better
         // integrated with Odoo (this refactoring should be done in master).
 
-        var def = this._super.apply(this, arguments);
-        if (this.$target.children('iframe').length) {
-            // There already is an <iframe/>, do nothing. This is the normal
-            // case. The whole code that follows is only there to ensure
-            // compatibility with videos added before bug fixes or new Odoo
-            // versions where the <iframe/> element is properly saved.
-            return def;
+        const proms = [this._super.apply(this, arguments)];
+        let iframeEl = this.$target[0].querySelector(':scope > iframe');
+
+        // The following code is only there to ensure compatibility with
+        // videos added before bug fixes or new Odoo versions where the
+        // <iframe/> element is properly saved.
+        iframeEl = iframeEl || this._generateIframe();
```

Comment:
In general I don't like this syntax, it probably has not execution effect but seeing `iframeEl = iframeEl` feels dubious. I always prefer an explicit
```javascript
if (!iframeEl) {
    iframeEl = ...
}
```
But agree, details.

Reply:
It would probably be dubious in a lot of other languages, but it is also idiomatic javascript, resulting from its out of control type coercion. Even in odoo it is common practice:

```
$ find odoo -type f -name '*.js' -exec grep -Hn "options = options ||" {} \+ | wc -l
87
```

In the end, a question of personal preference, and you'll be reading this code more often than me :wink: